src/dbnode/storage/index.go

```diff
@@ -1517,6 +1524,46 @@ func (i *nsIndex) CleanupExpiredFileSets(t time.Time) error {
 	return i.deleteFilesFn(filesets)
 }
 
+func (i *nsIndex) CleanupDuplicateFileSets() error {
+	fsOpts := i.opts.CommitLogOptions().FilesystemOptions()
+	infoFiles := i.readIndexInfoFilesFn(
+		fsOpts.FilePathPrefix(),
+		i.nsMetadata.ID(),
+		fsOpts.InfoReaderBufferSize(),
+	)
+	onDiskSegmentsOrderByVolumeIndexByVolumeType := make(map[idxpersist.IndexVolumeType][]fs.OnDiskSegments)
+	for _, file := range infoFiles {
+		seg := fs.NewOnDiskSegments(file.Info, file.AbsoluteFilepaths)
+		volumeType := seg.VolumeType()
+		if _, ok := onDiskSegmentsOrderByVolumeIndexByVolumeType[volumeType]; !ok {
+			onDiskSegmentsOrderByVolumeIndexByVolumeType[volumeType] = make([]fs.OnDiskSegments, 0)
+		}
+		onDiskSegmentsOrderByVolumeIndexByVolumeType[volumeType] = append(onDiskSegmentsOrderByVolumeIndexByVolumeType[volumeType], seg)
```

Comment:
Do we sort these by volume index? Or are they guaranteed by the index info files fn? Either way should either explicitly sort or mention how they're sorted. 

Given it's low volume I would even sort it again for good measure, to be defensive in case someone comes in later and makes read index info files not return the data ordered.